coldfront/core/user/views.py

```diff
@@ -546,28 +546,46 @@ def form_valid(self, form):
 
 class UserLoginView(View):
     """Redirect to the Basic Auth. login view or the SSO login view
-    based on enabled flags."""
+    based on enabled flags, retaining any provided next URL. If the user
+    is authenticated, redirect to the next URL or the home page."""
 
     def dispatch(self, request, *args, **kwargs):
-        basic_auth_enabled = 'BASIC_AUTH_ENABLED'
-        if flag_enabled(basic_auth_enabled):
-            return redirect(reverse('basic-auth-login'))
-        sso_enabled = 'SSO_ENABLED'
-        if flag_enabled(sso_enabled):
-            return redirect(reverse('sso-login'))
-        raise ImproperlyConfigured(
-            f'One of the following flags must be enabled: '
-            f'{basic_auth_enabled}, {sso_enabled}.')
+        next_url = request.GET.get('next')
+
+        if request.user.is_authenticated:
+            return redirect(next_url or reverse('home'))
+
+        basic_auth_enabled = flag_enabled('BASIC_AUTH_ENABLED')
+        sso_enabled = flag_enabled('SSO_ENABLED')
+        if not basic_auth_enabled ^ sso_enabled:
```

Comment:
Because of the XOR, this errors if both are enabled. That's intended behavior right?

Reply:
Yes, they're mutually exclusive.